client/src/App.tsx

```diff
@@ -485,143 +498,145 @@ const App = () => {
+                    resourceContent={resourceContent}
+                    nextCursor={nextResourceCursor}
+                    nextTemplateCursor={nextResourceTemplateCursor}
+                    error={errors.resources}
+                  />
+                  <PromptsTab
+                    prompts={prompts}
+                    listPrompts={() => {
+                      clearError("prompts");
+                      listPrompts();
+                    }}
+                    clearPrompts={() => {
+                      setPrompts([]);
+                      setNextPromptCursor(undefined);
+                    }}
+                    getPrompt={(name, args) => {
+                      clearError("prompts");
+                      getPrompt(name, args);
+                    }}
+                    selectedPrompt={selectedPrompt}
+                    setSelectedPrompt={(prompt) => {
+                      clearError("prompts");
+                      setSelectedPrompt(prompt);
+                    }}
+                    promptContent={promptContent}
+                    nextCursor={nextPromptCursor}
+                    error={errors.prompts}
+                  />
+                  <ToolsTab
+                    tools={tools}
+                    listTools={() => {
+                      clearError("tools");
+                      listTools();
+                    }}
+                    clearTools={() => {
+                      setTools([]);
+                      setNextToolCursor(undefined);
+                    }}
+                    callTool={(name, params) => {
+                      clearError("tools");
+                      callTool(name, params);
+                    }}
+                    selectedTool={selectedTool}
+                    setSelectedTool={(tool) => {
+                      clearError("tools");
+                      setSelectedTool(tool);
+                      setToolResult(null);
+                    }}
+                    toolResult={toolResult}
+                    nextCursor={nextToolCursor}
+                    error={errors.tools}
+                  />
+                  <ConsoleTab />
+                  <PingTab
+                    onPingClick={() => {
+                      void makeRequest(
+                        {
+                          method: "ping" as const,
+                        },
+                        EmptyResultSchema,
+                      );
+                    }}
+                  />
+                  <SamplingTab
+                    pendingRequests={pendingSampleRequests}
+                    onApprove={handleApproveSampling}
+                    onReject={handleRejectSampling}
+                  />
+                  <RootsTab
+                    roots={roots}
+                    setRoots={setRoots}
+                    onRootsChange={handleRootsChange}
+                  />
+                </div>
+              </Tabs>
+            </CapabilityContext.Provider>
```

Comment:
I think this approach works, but it feels a bit heavy right now. Instead of having each tab check the capabilities, let's instead set the default tab to be the first one that the server supports (right now we're defaulting to Resources and showing the error text in the PR screenshot). If none of the tabs are supported, let's just show some text in place of the tab content saying that the server doesn't support any MCP capabilities. We also shouldn't need the context at that point.